### PR TITLE
Validate sheet data

### PR DESCRIPTION
**Pull request by ipdae**
Introduce validate sheet data after `ISheet.Set`. it will be check sheet data without generic type.

### CODE REVIEW

File: .Lib9c.Tests/TableData/ISheetTest.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    public class ISheetTest
+    {
+        private const string Csv = @"id,round,arena_type,start_block_index,end_block_index,required_medal_count,entrance_fee,ticket_price,additional_ticket_price
+1,1,OffSeason,1,2,0,0,5,2
+1,2,Season,3,4,0,100,50,20
+1,3,OffSeason,5,1005284,0,0,5,2";
+
+        [Theory]
+        [InlineData(null, typeof(ArenaSheet))]
+        [InlineData(typeof(ArgumentException), typeof(BuffSheet))]
```
[the remark this answers is not in the document]
Comment (ipdae) [reply .Lib9c.Tests/TableData/ISheetTest.cs:16]: This test is check `Set` with a csv data not formatted from sheet type. not called `Validate`.

[the remark this answers is not in the document]
Comment (moreal) [reply .Lib9c.Tests/TableData/ISheetTest.cs:16]: Oops, it's my mistake 😞  Then, as I understood, in *Introduce validate sheet data after `ISheet.Set`.*, it expects the contributor manually writes a code to call the `ISheet.Validate()`, right?.

[the remark this answers is not in the document]
Comment (ipdae) [reply .Lib9c.Tests/TableData/ISheetTest.cs:16]: Right. if we call `Validate` in `Set` Hard forking is required. so I handled it this way.

[the remark this answers is not in the document]
Comment (moreal) [reply .Lib9c.Tests/TableData/ISheetTest.cs:16]: Yes, I also worried about it.  It seems good but it should be reviewed by the main contributors for sharing the pattern.

The last comments:

- The `Sheet.Validate()` checks only there is one row at least, right?
- Was there a situation the sheet was empty and occurred problems?
- I suggest using a new exception type like `SheetValidationException()`.

**Comment by ipdae** [comment]
I'll hard fork the PatchTableSheet and open pr again by working in the direction of calling Set in the relevant API.

**Pull request closed without merging**